avocado/plugins/spawners/lxc.py

```diff
@@ -0,0 +1,200 @@
+import asyncio
+import os
+import shutil
+import stat
+import tempfile
+
+try:
+    import lxc
+    LXC_AVAILABLE = True
+except ImportError:
+    LXC_AVAILABLE = False
+
+from avocado.core.output import LOG_JOB
+from avocado.core.plugin_interfaces import Init, Spawner
+from avocado.core.settings import settings
+from avocado.core.spawners.common import SpawnerMixin, SpawnMethod
+
+
+class LXCStreamsFile:
+
+    def __init__(self):
+        self.fd = None
+        self.path = None
+
+    def fileno(self):
+        return self.fd
+
+    def read(self):
+        with open(self.path, "r") as fp:
+            return fp.read()
+
+    def __enter__(self):
+        self.fd, self.path = tempfile.mkstemp()
+        return self
+
+    def __exit__(self, *args):
+        os.remove(self.path)
+
+
+class LXCSpawnerInit(Init):
+
+    description = 'LXC (container) based spawner initialization'
+
+    def initialize(self):
+        section = 'spawner.lxc'
+
+        help_msg = 'Distro for the LXC container'
+        settings.register_option(
+            section=section,
+            key='dist',
```

Comment:
Any reason for saving two letters here? My suggestions would be to use `distro`.

Reply:
Not in particular, the name was mostly inherited from LXC's `--dist` command line argument. Will update it to distro as we don't have to follow their naming.